tests/replay/test_dump.py

```diff
@@ -35,13 +35,13 @@ def fin_remove_replay_file() -> None:
 def test_type_error_if_no_template_name(replay_test_dir, context) -> None:
     """Test that replay.dump raises if the template_name is not a valid str."""
     with pytest.raises(TypeError):
-        replay.dump(replay_test_dir, None, context)
+        replay.dump(replay_test_dir, None, context)  # type: ignore[arg-type]
 
 
 def test_type_error_if_not_dict_context(replay_test_dir, template_name) -> None:
     """Test that replay.dump raises if the context is not of type dict."""
     with pytest.raises(TypeError):
-        replay.dump(replay_test_dir, template_name, 'not_a_dict')
+        replay.dump(replay_test_dir, template_name, 'not_a_dict')  # type: ignore[arg-type]
```

Comment:
i personally feel type safety makes this test (and ones like it) redundant. @jensens?

Reply:
I am fine removing these tests in favor of type checks.